src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs

```diff
@@ -784,6 +785,34 @@ static private int GetConsoleWindowWidth(int columnNumber)
             return columnNumber;
         }
 
+        /// <summary>
+        /// Return the console height.null  If not available (like when remoting), treat as Int.MaxValue.
+        /// </summary>
+        private static int GetConsoleWindowHeight()
+        {
+            if (InternalTestHooks.SetConsoleHeightToZero)
+            {
+                return DefaultConsoleHeight;
+            }
+
+            if (_noConsole)
+            {
+                return DefaultConsoleHeight;
+            }
+
+            try
+            {
+                // if Console height is set to 0, the default height is returned.
+                // This can happen in environments where TERM is not set.
+                return (Console.WindowHeight != 0) ? Console.WindowHeight : DefaultConsoleHeight;
```

Comment:
`_lo.RowNumber` works differently when output is redirected.  When piped to `less`, `RowNumber` is -1.  Only when the output is not piped does it have the correct number of on screen rows.  Changing that may break other things dependent on that behavior.  Similar for `_lo.ColumnNumber` where on redirection it becomes `int.MaxValue`.

I can do something similar to `GetConsoleWindowWidth()` where if the value is > 0, use it, otherwise get the real value from the console api.